ext/openssl/xp_ssl.c

```diff
@@ -2467,6 +2467,16 @@ static int php_openssl_sockop_set_option(php_stream *stream, int option, int val
 					/* fall through */
 					break;
 			}
+			break;
+
+		case STREAM_XPORT_OP_SHUTDOWN:
+			if (sslsock->ssl_active && (xparam->how == STREAM_SHUT_WR || xparam->how == STREAM_SHUT_RDWR)) {
+				if (SSL_shutdown(sslsock->ssl_handle) == -1) {
+					php_stream_socket_ops.set_option(stream, option, value, ptrparam); /* ensure socket shutdown either way, but report failure */
```

Comment:
Is this actually going to work for non-blocking socket? The thing is that if it is -1 then in case of the non-blocking socket the operation is not completed and it will have to be finished once the data can be read or written. It basically depends on `SSL_get_error` result. So closing socket might not be a good idea maybe.